src/Search.php

```diff
@@ -7281,56 +7281,91 @@ public static function giveItem(
-                        ]
-                    );
-                }
-            }
+        $aggregate = (isset($so['aggregate']) && $so['aggregate']);
+
+        $append_specific = static function ($specific, $field_data, &$out) use ($so) {
             if (!empty($specific)) {
                 $out .= $specific;
+            } else if (isset($field_data['values'])) {
+                // Aggregate values; No special handling
+                return;
             } else {
                 if (
                     isset($so['toadd'])
-                    && isset($so['toadd'][$data[$ID][$k]['name']])
+                    && isset($so['toadd'][$field_data['name']])
                 ) {
-                    $out .= $so['toadd'][$data[$ID][$k]['name']];
+                    $out .= $so['toadd'][$field_data['name']];
                 } else {
-                   // Empty is 0 or empty
+                    // Empty is 0 or empty
                     if (empty($split[0]) && isset($so['emptylabel'])) {
                         $out .= $so['emptylabel'];
                     } else {
-                       // Trans field exists
-                        if (isset($data[$ID][$k]['trans']) && !empty($data[$ID][$k]['trans'])) {
-                            $out .= $data[$ID][$k]['trans'];
+                        // Trans field exists
+                        if (isset($field_data['trans']) && !empty($field_data['trans'])) {
+                            $out .= $field_data['trans'];
                         } else {
-                            $value = $data[$ID][$k]['name'];
+                            $value = $field_data['name'];
                             $out .= $value !== null && $so['field'] === 'completename'
                                 ? CommonTreeDropdown::sanitizeSeparatorInCompletename($value)
                                 : $value;
                         }
                     }
                 }
             }
+        };
+        if (isset($table)) {
+            $itemtype = getItemTypeForTable($table);
+            if ($item = getItemForItemtype($itemtype)) {
+                if ($aggregate) {
+                    $tmpdata = [
+                        'values'     => [],
+                    ];
+                    foreach ($data[$ID] as $k => $v) {
+                        if (is_int($k)) {
+                            $tmpdata['values'][$k] = $v;
+                        } else {
+                            $tmpdata[$k] = $v;
+                        }
+                    }
+                    $specific = $item::getSpecificValueToDisplay(
+                        $field,
+                        $tmpdata,
+                        [
+                            'html'      => true,
+                            'searchopt' => $so,
+                            'raw_data'  => $data
+                        ]
+                    );
+
+                    $append_specific($specific, $tmpdata, $out);
```

Comment:
I do not get why call this method, as it cannot do more than appending `$specific`. Thus, adding the `else if (isset($field_data['values']))` block would not be required if appending is done directly here.
```php
            if (!empty($specific)) {
                $out .= $specific;
            } else if (isset($field_data['values'])) {
                // Aggregate values; No special handling
                return;
            }
```

Reply:
If no specific value is returned from `getSpecificValueToDisplay`, there was fallback logic here that doesn't apply for aggregated search options. Confusingly, if this code is reached, it doesn't necessarily mean that the search option datatype was set to `specific`; only that it wasn't one of the other types. In fact, I don't see the `specific` datatype referenced at all in the search engine, but that isn't really important here.

The alternative is to check if the return from `getSpecificValueToDisplay` is empty before calling the function in the `if ($aggregate) {` block instead of having that logic inside the function itself.